src/unix/signal.c

```diff
@@ -167,6 +172,10 @@ static void uv__signal_handler(int signum) {
 
     if (r != -1)
       handle->caught_signals++;
+
+    if ((handle->flags & UV_SIGNAL_ONE_SHOT) &&
+        (uv__signal_first_handle(handle->signum) == NULL))
+      uv__signal_unregister_handler(signum);
```

Comment:
Thanks for the hint, didn't know about it.

One general question though: how should multiple signal watchers listening on the same signal should be treated wrt this change? My initial idea was only unregistering the signal handler if no there watchers were left, but I don't know how useful that would be. Another option could be not allowing `uv_signal_start()` on a signal if 
`uv_signal_start_one_shot()` was already used and vice versa.